static/js/src/dynamic-forms.js

```diff
@@ -220,6 +238,7 @@ import { prepareInputFields } from "./prepare-form-inputs.js";
 
       contactModal.addEventListener("submit", function (e) {
         addLoadingSpinner();
+        setDataLayerConsentInfo();
```

Comment:
```suggestion
        setDataLayerConsentInfo(e.target);
```
You potentially pass the `e.target` into the function to find the form. The submit button _should_ always be inside the form element